app/MasterTemplate.hs

```diff
@@ -43,15 +43,16 @@ header page =
         H.ul ! A.id "nav-links" $ do
             H.li $ toLink "/graph" "Graph"
             H.li $ toLink "/grid" "Grid"
-            if page `elem` ["graph", "grid"]
-            then H.li $ H.a ! A.id "nav-export" $ "Export"
-            else ""
             H.li $ toLink "/generate" "Generate (beta)"
             -- H.li $ toLink "/timesearch" "Search"
             -- H.li $ toLink "/draw" "Draw"
             -- TODO: re-enable after handling new first-year courses
             -- H.li $ toLink "post" "Check My POSt!"
             H.li $ toLink "/about" "About"
+        if page `elem` ["graph", "grid"]
+        then H.button ! A.id "export" $ do
```

Comment:
The Haskell changes are indeed correct, but you should keep the old id (`nav-export`). It still fits, and in general it's best to avoid unnecessary changes.